grpc/src/main/java/com/replica/RouterImpl.java

```diff
@@ -223,6 +223,14 @@ public void routePt(PtRouteRequest request, StreamObserver<PtRouteReply> respons
         Point toPoint = request.getPoints(1);
         Request ghPtRequest = RouterConverters.toGHPtRequest(request);
 
+        // Set access and egress leg modes if they've been explicitly provided.
+        // Note: even if modes other than walk are requested, Graphhopper will return these legs
+        // as Trip.WalkLeg objects
+        String accessMode = request.getAccessMode().equals("") ? "foot" : request.getAccessMode();
+        String egressMode = request.getEgressMode().equals("") ? "foot" : request.getEgressMode();
+        ghPtRequest.setAccessProfile(accessMode);
```

Comment:
probably worth a comment describing the GH limitation and why we can specify a "mode" as a "profile" here

Reply:
Good call, added!